// angular-resize-event/esm2015/lib/angular-resized-event.module.js
import { NgModule } from '@angular/core';
import { ResizedDirective } from './resized.directive';
import * as ɵngcc0 from '@angular/core';
export class AngularResizedEventModule {
}
AngularResizedEventModule.ɵfac = function AngularResizedEventModule_Factory(t) { return new (t || AngularResizedEventModule)(); };
AngularResizedEventModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: AngularResizedEventModule });
AngularResizedEventModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularResizedEventModule, [{
        type: NgModule,
        args: [{
                declarations: [ResizedDirective],
                imports: [],
                exports: [ResizedDirective]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AngularResizedEventModule, { declarations: function () { return [ResizedDirective]; }, exports: function () { return [ResizedDirective]; } }); })();
